README.md

```diff
@@ -1,113 +1,132 @@
+. ~/Programming/robotics-prototype/robot/rospackages/devel/setup.bash
+. ~/Programming/robotics-prototype/venv/bin/activate
+source ~/Programming/robotics-prototype/robot/basestation/config/.bash_aliases
+```
+**Open a new terminal for changes to apply**. You should automatically have a virtual environment activated. The last line added a couple of aliases. You can read the .bash_aliases file to see all the new aliases.
 
-Finally, to link ROS to our webpage we need to install rosbridge: `sudo apt install ros-kinetic-rosbridge-suite`.
-Before running the GUI with `./app.py`, you need to run in a separate terminal: `roslaunch rosbridge_server rosbridge_websocket.launch`. Make sure to deactivate the `venv` for this terminal (otherwise it will crash).
-Make sure it works fine: after running the `./app.py` go visit `localhost:5000` in your browser (recommended browser: Google Chrome).
-For increased modularity, make sure to add to your `.bashrc` a line to source the `.bash_aliases` file in `robot/basestation/config/` so that you can use our shortcut aliases, such as `startgui` which you should always use to start the gui. It will automatically update the environment file for the browser called `env.js`.
+### Setup env.js
+You will need to provide the IP that the GUI will work with. By running this script, everything is handled.
 
-### .bashrc edits
-You should edit your `~/.bashrc` file so that it looks like this:
+```
+./robot/basestation/env.sh > robot/basestation/static/js/env.js
+```
+
+### Run the GUI
+Firstly, in a new terminal you should run `rosgui` (this is one of the new aliases in .bash_aliases that you added) to launch a ROS server
+
+Back in the original terminal, you can make use of the `base` alias, to be automatically directed to the basestation directory, and then run the app.py file.
 
 ```
-source /opt/ros/kinetic/setup.bash
-source ~/catkin_ws/devel/setup.bash
-# local mode
-#export ROS_MASTER_URI=http://localhost:11311
-#export ROS_HOSTNAME=localhost
-# competition mode
-export ROS_MASTER_URI=http://172.16.1.30:11311 # competition mode
-export ROS_HOSTNAME=beep
-. /home/beep/Programming/robotics-prototype/robot/rospackages/devel/setup.bash
-. /home/beep/Programming/robotics-prototype/venv/bin/activate
+$ base
+$ python app.py
 ```
+Alternatively, after running `rosgui` you can run `startgui` to run the GUI.
```

Comment:
It should be noted that running the GUI through `startgui` will automatically run the `updateEnv` alias too, which ensures that `env.js` is setup

Reply:
@PeterGhimself oh shit I didn't even know this lol, that's why I added

```
./robot/basestation/env.sh > robot/basestation/static/js/env.js
```